suite-native/module-accounts-management/src/components/AccountDetailCryptoValue.tsx

```diff
@@ -1,26 +1,46 @@
 import { memo } from 'react';
 
 import { HStack } from '@suite-native/atoms';
-import { CryptoAmountFormatter } from '@suite-native/formatters';
+import { CryptoAmountFormatter, EthereumTokenAmountFormatter } from '@suite-native/formatters';
 import { CryptoIcon } from '@suite-common/icons';
 import { NetworkSymbol } from '@suite-common/wallet-config';
+import { TokenAddress, TokenSymbol } from '@suite-common/wallet-types';
 
 type AccountDetailBalanceProps = {
     value: string;
     networkSymbol: NetworkSymbol;
     isBalance?: boolean;
+    decimals?: number;
+    tokenSymbol?: TokenSymbol;
+    tokenAddress?: TokenAddress;
 };
 
 export const AccountDetailCryptoValue = memo(
-    ({ value, networkSymbol, isBalance = true }: AccountDetailBalanceProps) => (
+    ({
+        value,
+        networkSymbol,
+        tokenSymbol,
+        tokenAddress,
+        isBalance = true,
+        decimals,
+    }: AccountDetailBalanceProps) => (
         <HStack spacing="small" flexDirection="row" alignItems="center" justifyContent="center">
-            <CryptoIcon symbol={networkSymbol} size="extraSmall" />
-            <CryptoAmountFormatter
-                value={value}
-                network={networkSymbol}
-                isBalance={isBalance}
-                adjustsFontSizeToFit
-            />
+            <CryptoIcon symbol={tokenAddress || networkSymbol} size="extraSmall" />
+            {tokenSymbol ? (
+                <EthereumTokenAmountFormatter
+                    value={value}
+                    symbol={tokenSymbol}
+                    adjustsFontSizeToFit
+                />
+            ) : (
+                <CryptoAmountFormatter
+                    value={value}
+                    network={networkSymbol}
+                    isBalance={isBalance}
+                    adjustsFontSizeToFit
+                    decimals={decimals}
```

Comment:
Sadly it doesn't work because when you use token balance from redux it's already formatted, but if you do calculations locally there are no decimals so you need to add them later.

This kind of sucks, but this is probably best solution that doesn't require big refactors.

Reply:
I still don't understand. This PR should only extend the functionality by displaying graph for token details, right?

So why is modified usage of `CryptoAmountFormatter` that should be rendered only on the mainnet (ETH, BTC etc.) screens? In case of token detail screen, only the `EthereumTokenAmountFormatter` should be rendered right?

So my question is: "Why is needed modification of the original code?"

Reply:
You are right. In the end I removed it entirely in 19fd4da132 because it now seems to work everywhere without it.